tests/runtimes/test_run.py

```diff
@@ -76,6 +76,30 @@ def test_new_function_from_runtime():
     )
 
 
+def test_new_function_default_foo():
+    runtime = _get_runtime()
+    function = mlrun.new_function(runtime=runtime)
+    expected_runtime = runtime
+    expected_runtime["spec"]["foo"] = mlrun.mlconf.function.spec.foo.default
+    assert function.spec.foo == mlrun.mlconf.function.spec.foo.default
+
+
+def test_new_function_override_default_foo():
+    mlrun.mlconf.function.spec.foo.default = "new_default"
+    runtime = _get_runtime()
+    function = mlrun.new_function(runtime=runtime)
+    expected_runtime = runtime
+    expected_runtime["spec"]["foo"] = mlrun.mlconf.function.spec.foo.default
```

Comment:
it is not very clear what how changing `expected_runtime` affects the function, perhaps add a comment on what you are doing here.

Reply:
Since it's a dummy property, I believe the only expected effect would be the new default being correctly inherited from the defaults dict.

What did you have in mind for this comment?

Reply:
I guess just that changing the runtime this way also affects the function spec itself, because it is passed by reference (if that is indeed the case).

Reply:
I see, that's a great point!

However, we might not need to worry about the main runtime pointer in this specific case, since `new_function` does a [deepcopy of the runtime when it's defined as a dictionary](https://github.com/mlrun/mlrun/blob/development/mlrun/run.py#L601-L602) (I assume it's done to avoid this kind of unintentional change).

What do you think? We might prefer to drop this test altogether if it doesn't really behave as we intend it to.